cheroot/server.py

```diff
@@ -1204,6 +1254,190 @@ def send_headers(self):
+
+    ready = False
+    """When True, the request has been parsed and is ready to begin generating
+    the response. When False, signals the calling Connection that the response
+    should not be generated and the connection should close."""
+
+    close_connection = False
+    """Signals the calling Connection that the request should close. This does
+    not imply an error! The client and/or server may each request that the
+    connection be closed."""
+
+    chunked_write = False
+    """If True, output will be encoded with the "chunked" transfer-coding.
+
+    This value is set automatically inside send_headers."""
+
+    header_reader = HeaderReader()
+    """
+    A HeaderReader instance or compatible reader.
+    """
+
+    def __init__(self, server, conn, proxy_mode=False, strict_mode=True):
+        """Initialize HTTP request container instance.
+
+        Args:
+            server (HTTPServer): web server object receiving this request
+            conn (HTTPConnection): HTTP connection object for this request
+            proxy_mode (bool): whether this HTTPServer should behave as a PROXY
+            server for certain requests
+            strict_mode (bool): whether we should return a 400 Bad Request when
+            we encounter a request that a HTTP compliant client should not be
+            making
+        """
+        self.server = server
+        self.conn = conn
+        self.h_conn = h11.Connection(h11.SERVER)
+        self.ready = False
+        self.started_request = False
+        self.scheme = b'http'
+        if self.server.ssl_adapter is not None:
+            self.scheme = b'https'
+        # Use the lowest-common protocol in case read_request_line errors.
+        self.response_protocol = 'HTTP/1.0'
+        self.inheaders = {}
+
+        self.status = ''
+        self.outheaders = []
+        self.sent_headers = False
+        self.close_connection = self.__class__.close_connection
+        self.chunked_read = False
+        self.chunked_write = self.__class__.chunked_write
+        self.proxy_mode = proxy_mode
+        self.strict_mode = strict_mode
+
+    def _next_event(self):
+        while True:
+            event = self.h_conn.next_event()
+            if event is h11.NEED_DATA:
+                if self.h_conn.they_are_waiting_for_100_continue:
+                    go_ahead = h11.InformationalResponse(status_code=100)
+                    bytes_out = self.h_conn.send(go_ahead)
+                    self.conn.wfile.write(bytes_out)
+                self.h_conn.receive_data(self.conn.rfile.readline())
+                continue
+            return event
+
+    def parse_request(self):
+        """Parse the next HTTP request start-line and message-headers."""
+        req_line = self._next_event()
+        if isinstance(req_line, h11.Request):
+            self.started_request = True
+            self.uri = req_line.target
+            scheme, netloc, path, query, fragment = urllib.parse.urlsplit(self.uri)
```

Comment:
I think you could save a few lines by doing the assignments directly
```suggestion
            _scheme, _netloc, self.path, self.qs, _fragment = urllib.parse.urlsplit(self.uri)
```